packages/dds/tree/docs/storage/compatibility.md

```diff
@@ -0,0 +1,273 @@
+# Compatibility
+
+This document provides concrete guidelines and strategies for organizing code that impacts SharedTree's persisted format.
+It then pivots to the current state of SharedTree's persisted format, and outlines steps to take to align the implementation
+with those guidelines.
+
+## Prerequisites
+
+[This document](../../../SchemaVersioning.md) provides general "best practices" for working with persisted data within the Fluid Framework.
+It's strongly recommended to read through and understand its rationale before continuing with this document,
+as most of the concrete recommendations presented henceforth fall out of those best practices.
+
+## What State is Persisted?
+
+A DDS's persisted format encompasses the format it uses for its summaries as well as its ops (due to [trailing ops](../../../README.md))
+including transitively referenced structured blob data.
+
+Since documents are stored outside of Fluid control (i.e. no type of central data migration is possible),
+DDSes necessarily commit to backwards compatibility of their format for all time.
+
+## Format Management
+
+The actively used format is stored and managed by an `IPersistedConfigStore`.
```

Comment:
where can I find IPersistedConfigStore? Seems to not exists in FF repo

Reply:
I have a working but rough prototype in #14632. As in, it's fuzz tested and implements a mechanism which aids migration, but API/code quality I'm not yet happy with. That's part of the reason this doc isn't yet checked in, another being I'm investigating using the DDS' `.attributes` blob to contain this sort of data.